src/solvers/topDown_deprecated.ml

```diff
@@ -18,20 +18,15 @@ module TD2 =
 
     module VI =
     struct
-      type t = Var.t * int
-      let compare (x,n) (y,m) =
-        match compare n m with
-        | 0 -> Var.compare x y
-        | n -> n
+      type t = Var.t * int [@@deriving ord]
```

Comment:
Was this a deliberate choice to call compare on `x` and `y` only in case `n <> m`? One could imagine this would be more efficient?

Reply:
Good catch! Although I wouldn't bother changing it: that's inside a deprecated `TD2` solver, which at the end of the file isn't even registered. Even if it was some kind of optimization, I think it's irrelevant because TD3 doesn't use such thing anyway.

Reply:
I decided to revert this one along with the other place I found, just in case. And I added comments pointing this out to avoid anyone else tempting this change.